packages/protobuf/src/create.ts

```diff
@@ -260,6 +262,26 @@ function createZeroMessage(desc: DescMessage): Message {
   return msg as Message;
 }
 
+/**
+ * Do we need the prototype chain to track field presence?
+ */
+function needsPrototypeChain(desc: DescMessage): boolean {
+  switch (desc.file.edition) {
+    case EDITION_PROTO3:
+      // proto3 always uses implicit presence, we never need the prototype chain.
+      return false;
+    case EDITION_PROTO2:
+      // proto2 never uses implicit presence, we always need the prototype chain.
+      return true;
+    default:
+      // If a message uses scalar or enum fields with explicit presence, we need
+      // the prototype chain to track presence. This rule does not apply to fields
+      // in a oneof group - they use a different mechanism to track presence.
+      return desc.fields.some(
+        (f) => f.presence != IMPLICIT && f.fieldKind != "message" && !f.oneof,
+      );
+  }
+}
```

Comment:
Now we avoid the prototype chain for messages that only use implicit presence. Message fields, oneof, lists, and maps never need the prototype chain.

We're keeping the cases for proto2 and proto3 because this is a hot path.